### PR TITLE
9735 SP/GP Create UI - Staff payment and update certify component

### PR DESCRIPTION
**Pull request by lewischenstudio**
Issue #: [/bcgov/entity#9735](https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/bcgov/entity/9735)

*Description of changes:*
- ~insert staff payment component~ _already done_
- ~editable completing party in documents delivery~ _already done_
- ~update certify section to be third party, not the "I, ___ certify"~ _already done_
- [x] Added "My Business Registry" in the breadcrumb (with reference to [Invision](https://projects.invisionapp.com/share/3511UPXPMT2P#/screens/464799355))
- [x] Display Business Number for Sole Proprietors

Comment: Additional requirement to redirect "My Business Registry" to the list of businesses page with staff login.
Suggestion: new SBC AUTH ticket to work on the redirect url.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Registration/DefineRegistrationSummary.vue [hunk 1]
```diff
@@ -85,7 +85,7 @@
       </template>
 
       <!-- Business Number -->
-      <template v-if="isTypePartnership">
+      <template>
```
Comment (severinbeauvais) [inline src/components/Registration/DefineRegistrationSummary.vue:88]: Just delete this pseudo-element then. Vue uses it for conditional logic only -- it doesn't render.

File: src/resources/BreadCrumbResource.ts [hunk 2]
```diff
@@ -66,3 +66,11 @@ export function getStaffDashboardBreadcrumb (): BreadcrumbIF {
     href: `${sessionStorage.getItem('BUSINESSES_URL')}staff/${getParams()}`
   }
 }
+
+export function getStaffRegistryBreadcrumb (): BreadcrumbIF {
+  const accountId = sessionStorage.getItem('ACCOUNT_ID')
```
Comment (severinbeauvais) [inline src/resources/BreadCrumbResource.ts:71]: This should be handled by `navigate()` instead.

Comment (severinbeauvais) [reply src/resources/BreadCrumbResource.ts:71]: Oh wait, this is different.

Looks fine 👍

Comment (severinbeauvais) [reply src/resources/BreadCrumbResource.ts:71]: Actually, the ideal solution might be for Auth Web to redirect from, say:
`https://dev.bcregistry.ca/business/auth/account/business?accountid=2288`
to:
`https://dev.bcregistry.ca/business/auth/account/668/business`

(It partially does this already for non-staff/IDIR logins.)

Do you want to try looking into this?

File: src/resources/BreadCrumbResource.ts [hunk 3]
```diff
@@ -66,3 +66,11 @@ export function getStaffDashboardBreadcrumb (): BreadcrumbIF {
     href: `${sessionStorage.getItem('BUSINESSES_URL')}staff/${getParams()}`
   }
 }
+
+export function getStaffRegistryBreadcrumb (): BreadcrumbIF {
```
Comment (severinbeauvais) [inline src/resources/BreadCrumbResource.ts:70]: It's actually not the "Registry" page (which is https://www.bcregistry.gov.bc.ca/) but the "(My) Business Registry" page

**Comment by severinbeauvais** [comment]
Please un-check the items in the description that are incomplete or not part of this Pull Request.

Also, please update the app version so that you can track your change through dev/test/prod environments.

File: src/resources/BreadCrumbResource.ts (hunk 2 above)
Comment (lewischenstudio) [reply src/resources/BreadCrumbResource.ts:71]: Yes. I need to try the Auth Web. Because it didn't redirect properly with my way.

Comment (severinbeauvais) [reply src/resources/BreadCrumbResource.ts:71]: The reason I recommend a fix to Auth Web is because many UIs/web apps have this same breadcrumb feature, and all would need to be updated now and any time the URL changes. (The URL is otherwise provided by external config, so no rebuilding needed.)

If the problem is that Auth Web isn't going to the right business page when the user is staff/IDIR, then that should be fixed directly.

OTOH, if the PO decides to accept this temporary fix now, and we have a ticket to implement a proper fix and delete this workaround later, then I'm OK with this.

Comment (lewischenstudio) [reply src/resources/BreadCrumbResource.ts:71]: This is a good question. Yui and Tracy did suggest to work on this fix in another ticket. I can now see how this has an impact on other UIs, such as filing-ui that doesn't have "My Business Registry" either (on staff). Meanwhile, I am not sure what I need to do with Auth Web. If "My Business Registry" should appear in all UIs, then it's better to work on it in another ticket and discard any change in this ticket.

File: src/components/Registration/DefineRegistrationSummary.vue [hunk 4]
```diff
@@ -85,20 +85,18 @@
       </template>
 
       <!-- Business Number -->
-      <template v-if="isTypePartnership">
-        <v-divider class="mx-6" />
+      <v-divider class="mx-6" />
```
Comment (thorwolpert) [inline src/components/Registration/DefineRegistrationSummary.vue:88]: It doesn't look like any change here, other than changing indenting.
please don't make changes to formatting, for the sake of changing formatting as it detracts from understanding what the PR is attempting to do.

**Review by thorwolpert — COMMENTED** [review]
Please don't make changes to formatting for the sake of changing formatting.

File: src/components/Registration/DefineRegistrationSummary.vue (hunk 4 above)
Comment (severinbeauvais) [reply src/components/Registration/DefineRegistrationSummary.vue:88]: Looks like a conditional was removed, so I'm OK with fixing the indentation of the block.

File: src/components/Registration/DefineRegistrationSummary.vue [hunk 5]
```diff
@@ -85,20 +85,18 @@
       </template>
 
       <!-- Business Number -->
-      <template v-if="isTypePartnership">
```
Comment (severinbeauvais) [inline src/components/Registration/DefineRegistrationSummary.vue:88]: Wait - why did you delete this?

File: src/resources/BreadCrumbResource.ts (hunk 2 above)
Comment (severinbeauvais) [reply src/resources/BreadCrumbResource.ts:71]: What's the other ticket? Perhaps this needs a larger discussion... please raise it at Standup. Thx.

If we take the breadcrumbs out of this ticket, does THIS ticket need to change any code?

Comment (lewischenstudio) [reply src/resources/BreadCrumbResource.ts:71]: If we take the breadcrumbs out of this ticket, then this ticket won't have any change.

File: src/components/Registration/DefineRegistrationSummary.vue (hunk 5 above)
Comment (lewischenstudio) [reply src/components/Registration/DefineRegistrationSummary.vue:88]: According to the [Invision design](https://projects.invisionapp.com/share/3511UPXPMT2P#/screens/464799355), SP also displays BN, so I removed this condition.

File: src/resources/BreadCrumbResource.ts (hunk 2 above)
Comment (severinbeauvais) [reply src/resources/BreadCrumbResource.ts:71]: When I am logged in as staff (IDIR), the following URL doesn't redirect to My Business Registry. Do I have the right URL?
https://dev.bcregistry.ca/business/auth/account/668/business

Comment (severinbeauvais) [reply src/resources/BreadCrumbResource.ts:71]: Ohhhhh, I think I know!

If staff start from the [staff dashboard[(https://dev.bcregistry.ca/business/auth/staff/dashboard/active) and go the Entity Dashboard or Manage Account pages then a bunch of account-related fields are set in session storage and those pages display correctly.

However, if staff directly go to a business' dashboard (eg, https://dev.bcregistry.ca/business/FM1000007) then those session variables are not set the and going to My Business Registry (eg, https://dev.bcregistry.ca/business/auth/account/668/business) doesn't work.

All this to say this is more complicated than initially believed.

And also this change has been removed from this ticket.

Comment (lewischenstudio) [reply src/resources/BreadCrumbResource.ts:71]: Yes, you are right. "My Business Registry" on the breadcrumbs can redirect to either of the two pages: staff's registry page or the user's registry page. Suppose we have it in the breadcrumbs for both cases. 
1. If a staff is accessing a user's NR and wants to return to the user's registry page, then this button should do that. 
2. If a staff is accessing his/her own NR and wants to return his/her own registry page, then this button also needs to do that. 
This button is very complicated and I feel sorry to bring it up. 

Since BN isn't required for SP, there isn't any change at all in this branch. 

I think it's safe to close this pull request.

Comment (severinbeauvais) [reply src/resources/BreadCrumbResource.ts:71]: Agreed. I'm resolving this conversation.

File: src/components/Registration/DefineRegistrationSummary.vue (hunk 4 above)
Comment (severinbeauvais) [reply src/components/Registration/DefineRegistrationSummary.vue:88]: This change is obsolete as we want to continue displaying BN only for GPs.
I'm resolving this conversation.

File: src/resources/BreadCrumbResource.ts (hunk 3 above)
Comment (severinbeauvais) [reply src/resources/BreadCrumbResource.ts]: Obsolete. Resolving.

File: src/components/Registration/DefineRegistrationSummary.vue (hunk 5 above)
Comment (severinbeauvais) [reply src/components/Registration/DefineRegistrationSummary.vue:88]: This change is obsolete as we want to continue displaying BN only for GPs.
Resolving.

**Comment by lewischenstudio** [comment]
- Items in the Todo list are already implemented. 
- "My Business Registry" are not required for this ticket. 
- BN is not required for SP. 

I'm closing this branch since there is no change.
